client/dfclient/src/loading_state.cpp

```diff
@@ -0,0 +1,164 @@
+#include <functional>
+#include <iostream>
+
+#include <ncine/Application.h>
+#include <ncine/FileSystem.h>
+#include <ncine/MeshSprite.h>
+#include <ncine/Sprite.h>
+#include <ncine/Texture.h>
+
+#include "loading_state.hpp"
+#include "resources.hpp"
+
+namespace nc = ncine;
+
+inline long fileSize(const char* path) { return path == "" ? 0 : ncine::FileSystem::fileSize(path); }
+inline long fileSize(std::string path) { return path == "" ? 0 : ncine::FileSystem::fileSize(path.c_str()); }
+
+LoadingState::LoadingState(Resources& r) : _resources(r) {
+	nc::SceneNode &rootNode = nc::theApplication().rootNode();
+	auto res = nc::theApplication().appConfiguration().resolution;
+	nc::theApplication().gfxDevice().setClearColor(nc::Colorf::Black);
+
+	// iterate over files to load, calculating the total size and registering file names
+	// init paths
+	auto rootPath = std::string(ncine::theApplication().appConfiguration().dataPath().data());
+	auto textureDir = ncine::FileSystem::Directory((rootPath + std::string(TEXTURES_PATH)).c_str());
+	auto soundDir = ncine::FileSystem::Directory((rootPath + SOUNDS_PATH).data());
+
+	// register everything
+	const char* file = textureDir.readNext();
+	while (file)
+	{
+		auto absPath = rootPath + std::string(TEXTURES_PATH) + "/" + std::string(file);
+		if (ncine::FileSystem::isFile(absPath.c_str())) {
+			_total += fileSize(absPath);
+			_textureFilenames.push_back(absPath);
+		}
+		file = textureDir.readNext();
+	}
+	textureDir.close();
+
+	file = soundDir.readNext();
+	while (file) {
+		auto absPath = rootPath.data() + std::string(SOUNDS_PATH) + "/" + std::string(file);
+		if (ncine::FileSystem::isFile(absPath.c_str())) {
+			_total += fileSize(absPath);
+			_soundFilenames.push_back(absPath);
+		}
+		file = soundDir.readNext();
+	}
+
```

Comment:
the textureDir is closed, but soundDir never is. does that make any difference? please decide on one of the variants

Reply:
That's right, my bad. Added